classes/clean.php

```diff
@@ -352,6 +352,30 @@ protected static function get_courses($criteria = array()) {
         return $DB->get_records_select_menu('course', 'id > 1 '.$extrasql, $params, '', 'id, id');
     }
 
+    /**
+     * Execute SQL in single transaction
+     *
+     * @param  string $sql
+     */
+    protected static function execute_sql($sql) {
```

Comment:
What is the intent of this function? Why do we care about a transaction given that the data wash is run in a single process in serial anyway?

Reply:
You may want all or nothing. In case you don't it does not hurt.

Reply:
I don't think this is doing what you think it is. Moodle's sql api only allows 1 sql statement per execute call. I've just tested this with as the test sql which I think is valid and should work:

```
select count(*) as bar1 from {user};
select count(*) as bar2 from {user};
```

and this throws an exception. Can you please split up the spit on ; and run each. I've also turned the setting into a text area to make it easier to see the sql

Reply:
```brendan@zebrafish 09:44:25 /var/www/moodle.local/local/datacleaner $ php cli/clean.php  --run-pre-wash  -v --force
Safety checks skipped due to --force command line option.

== Running custom_sql_pre cleaner ==
Executing: select count(*) as bar1 from {user};
select count(*) as bar2 from {user};
Database transaction aborted automatically in /var/www/moodle.local/local/datacleaner/cli/clean.php
Default exception handler: Coding error detected, it must be fixed by a programmer: moodle_database::execute() Multiple sql statements found or bound parameters not used properly in query! Debug: 
Error code: codingerror
* line 703 of /lib/dml/pgsql_native_moodle_database.php: coding_exception thrown
* line 375 of /local/datacleaner/classes/clean.php: call to pgsql_native_moodle_database->execute()
* line 43 of /local/datacleaner/cleaner/custom_sql_pre/classes/clean.php: call to local_datacleaner\clean::execute_sql()
* line 145 of /local/datacleaner/cli/clean.php: call to cleaner_custom_sql_pre\clean::execute()

!!! Coding error detected, it must be fixed by a programmer: moodle_database::execute() Multiple sql statements found or bound parameters not used properly in query! !!!
!! 
Error code: codingerror !!
!! Stack trace: * line 703 of /lib/dml/pgsql_native_moodle_database.php: coding_exception thrown
* line 375 of /local/datacleaner/classes/clean.php: call to pgsql_native_moodle_database-&gt;execute()
* line 43 of /local/datacleaner/cleaner/custom_sql_pre/classes/clean.php: call to local_datacleaner\clean::execute_sql()
* line 145 of /local/datacleaner/cli/clean.php: call to cleaner_custom_sql_pre\clean::execute()

Reply:
> I don't think this is doing what you think it is. Moodle's sql api only allows 1 sql statement per execute call. I've just tested this with as the test sql which I think is valid and should work:
> 
> ```
> select count(*) as bar1 from {user};
> select count(*) as bar2 from {user};
> ```
> 
> and this throws an exception. Can you please split up the spit on ; and run each. I've also turned the setting into a text area to make it easier to see the sql

Will split, but that is strange. SELECT 1;\nSELECT 2 works in the test.